drivers/dma/dma_mcux_lpc.c

```diff
@@ -340,16 +353,60 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 	switch (config->channel_direction) {
 	case MEMORY_TO_MEMORY:
 		is_periph = false;
-		src_inc = 1;
-		dst_inc = 1;
+		if (block_config->source_gather_en) {
+			src_inc = block_config->source_gather_interval / width;
+			/* The current controller only supports incrementing the
+			 * source and destination up to 4 time transfer width
+			 */
+			if ((src_inc > 4) || (src_inc == 3)) {
```

Comment:
https://docs.zephyrproject.org/latest/doxygen/html/group__sys-util.html#ga52d277cbf33f76350b2fcb21c24640ee
`IS_POWER_OF_TWO` I am assuming that's what the check for 3 is

Reply:
The only values we allow are 1, 2, 4. The macro you provided will work as well. However checking for 3 in this case seems simpler.